src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -105,29 +115,67 @@ private void getMealById(HttpServletRequest request, HttpServletResponse respons
             response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
-
-        Meal meal = dishes.get(id);
-
-        if (meal != null) {
-            String gson = new Gson().toJson(meal);
-            response.setContentType("application/json");
-            response.getWriter().println(gson);
+        Filter propertyFilter = new FilterPredicate("id", FilterOperator.EQUAL, id);
+        Query query = new Query("Meal").setFilter(propertyFilter);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        HashMap<Long, Meal> meals = getDataFromDatastore(results); 
+        
+        if (meals.size() > 1) {
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             return;
         }
-
+        if (meals.size() == 1) {
+            Meal meal = meals.get(0L);
+            if (meal != null) {
+                String gson = new Gson().toJson(meal);
+                response.setContentType("application/json");
+                response.getWriter().println(gson);
+                return;
+            }
+        }
         response.sendError(HttpServletResponse.SC_NOT_FOUND);
         return;
     }
 
-     private void returnIdOfSimilar(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    private HashMap<Long, Meal> getDataFromDatastore(PreparedQuery query) {
+        HashMap<Long, Meal> result = new HashMap<>();
+        Long ind = 0L;
+        for (Entity entity : query.asIterable()) {
+            try {
+                Long id = (Long)entity.getProperty("id");
+                String title = (String) entity.getProperty("title");
+                String description = (String) entity.getProperty("description");
+                ArrayList<String> ingredients = (ArrayList<String>) entity.getProperty("ingredients");
+                String type = (String) entity.getProperty("type");
+                Meal meal = new Meal(id, title, description, ingredients, type);
+                result.put(ind, meal);
+            } catch(ClassCastException e) {
+                continue;
```

Comment:
You probably want to at least log the error to `System.out.print`.

In general, whenever an error-ish situation happens, it's better not to silently ignore it. Typically you would either:
1.  propagate the error, either in form of reraising the exception or by including it in the return value (depending on the language)
2.  explicitly handle the error, e.g. by replacing the failing operation with some default
3.  simply log the error message

I'd suggest using 3. here.